# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1361632136 18000
#      Sat Feb 23 10:08:56 2013 -0500
# Node ID def3b111cff86966b6d278fa8af3777685c49ed1
# Parent  1785493171ac416b918b7029975f9acb9121adad
avoid gcc warning about int to char conversion

* sysdep.cc (Fkbhit): Explicitly cast int to char.

diff --git a/libinterp/interpfcn/sysdep.cc b/libinterp/interpfcn/sysdep.cc
--- a/libinterp/interpfcn/sysdep.cc
+++ b/libinterp/interpfcn/sysdep.cc
@@ -650,17 +650,17 @@ returning the empty string if no key is 
     {
       Fdrawnow ();
 
       int c = octave_kbhit (args.length () == 0);
 
       if (c == -1)
         c = 0;
 
-      char s[2] = {c, '\0'};
+      char s[2] = { static_cast<char> (c), '\0' };
 
       retval = s;
     }
 
   return retval;
 }
 
 DEFUN (pause, args, ,
